src/js/pages/services/ServicesTab.js

```diff
@@ -108,6 +110,15 @@ var ServicesTab = React.createClass({
 
     // Find item in root tree and default to root tree if there is no match
     let item = DCOSStore.serviceTree.findItemById(id) || DCOSStore.serviceTree;
+    let groupId = ServicesPathUtils.getGroupFromAppId(item.id);
+
+    if (item instanceof ServiceTree) {
+      groupId = item.id;
+
+      if (groupId != null && !groupId.endsWith('/')) {
```

Comment:
This shouldn't be necessary and seems very error prone. Please refactor the breadcrump component to simplify it's usage.